### ISSUE TITLE
Kahoot.JS doesn't work on Node 10.19.0 and below

### ISSUE THREAD

**Issue by robigan** [theusaf/kahoot.js-updated#44]
So I am in version 1.4.24, I was testing to see if it worked after realizing that the original kahoot.js wasn't working, I nabbed this file to confirm if it worked (https://github.com/theusaf/kahoot.js-updated/blob/master/examples/basicUsage.js) and when I ran it it just printed connecting to kahoot and then "exited" the program, no error no nothing, note that I am in Linux mint 19.3, I tried this both in VSC and in the console. So what am I doing wrong? Its also not showing up in the kahoot am hosting, is this due to me hosting on the same IP?

**Comment by theusaf** [comment]
I just tested it on my computer. It works as intended. Make sure you replace the PIN with the actual kahoot pin that you are hosting for it to work. If the pin is invalid, the join method should reject with the message "token_error" and exit the program. Otherwise, it should create the event emitter.

**Comment by robigan** [comment]
No, nothing happened, even with a wrong pin

**Comment by theusaf** [comment]
What happens when you run this code?
```js
var Kahoot = require("kahoot.js-updated");
var client = new Kahoot;
console.log("Joining kahoot...");
client.join(9802345 /* Or any other kahoot token */, "kahoot.js").catch(e=>{
        console.log("Error: " + e);
});
client.on("joined", () => {
	console.log("I joined the Kahoot!");
});
client.on("quizStart", quiz => {
	console.log("The quiz has started! The quiz's name is:", quiz.name);
});
client.on("questionStart", question => {
	console.log("A new question has started, answering the first answer.");
	question.answer(0);
});
client.on("quizEnd", () => {
	console.log("The quiz has ended.");
});
```

**Comment by robigan** [comment]
As I already said, nothing happens, as seen in the image

**Comment by theusaf** [comment]
Well, nothing is supposed to happen if the PIN is wrong.

**Comment by robigan** [comment]
Yes and I fully understand, but as I mentioned there were no errors, yes there was one time when the pin was wrong but even then I rechecked everything and inputted the right pin, no error messages or anything else, as I already said, the “bot” didn’t even show up on the player list

**Comment by robigan** [comment]
Update: As I said there were some peer dependency issues, and this is still not functioning, so what I did was create a whole new directory, I then proceeded to install kahoot.js-updated, with the issue of the peer dependencies, so I installed those too, these were bufferutil and utf-8-validate, once all that was satisfied, I ran the test file recommended, still no change there

**Comment by theusaf** [comment]
What version of Node do you have? Also, are you installing any other packages besides kahoot.js-updated?

**Comment by theusaf** [comment]
I created a REPL: https://repl.it/@theusaf/KahootJSExample#index.js It seems to work as expected, not sure why it isn't working for you.

**Comment by robigan** [comment]
> What version of Node do you have? Also, are you installing any other packages besides kahoot.js-updated?
I am running node version 10.19.0

**Comment by robigan** [comment]
Update: I finally tried the repl, I have some things to say
A) FUCKING THANK YOU FOR USING REPL.IT ITS SOO GUD
B) Yes it works, maybe the node version?

**Comment by robigan** [comment]
Update 2: I cannot exactly confirm; but yes it looks like node version 10.19.0 is not compatible, after some learning about curl | sudo and adding software repos to my apt list (I am still new to Linux) I was able to upgrade to node 12.18.2 and it now works

**Comment by theusaf** [comment]
Thanks for confirming. I’ll see if I can figure out how to make it work with 10.19.0 and below if possible.

Sent over the interblag from a GUI.
https://github.com/theusaf
https://shortr.github.io/?llorkcir

**Comment by theusaf** [comment]
I was unable to replicate this issue using version 10.19.0 I can confirm that it doesn't work on version 8.17.0 and below. I will be updating README to reflect these changes.

**Comment by theusaf** [comment]
Closed in 108e609
